src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -57,12 +59,19 @@ internal sealed partial class CompletionHandler : ILspServiceDocumentRequestHand
             var completionOptions = GetCompletionOptions(document, capabilityHelper);
             var completionService = document.GetRequiredLanguageService<CompletionService>();
 
+            var isInitialRequest = request.Context?.TriggerKind is not LSP.CompletionTriggerKind.TriggerForIncompleteCompletions;
+
             // Let CompletionService decide if we should trigger completion, unless the request is for incomplete results, in which case we always trigger. 
-            if (request.Context?.TriggerKind is not LSP.CompletionTriggerKind.TriggerForIncompleteCompletions
+            if (isInitialRequest
                 && !completionService.ShouldTriggerCompletion(document.Project, document.Project.Services, documentText, position, completionTrigger, completionOptions, document.Project.Solution.Options, roles: null))
             {
                 return null;
             }
+            
+            const int TelemetryDelayThresholdMs = 500;
+            var metricName = isInitialRequest ? $"InitialRequest" : $"IncompleteRequest";
```

Comment:
I'm assuming the default roslyn LSP request telemetry not enough here because we need to know if this is the initial vs incomplete request?

(Note that for VS as long as the first invocation had more than 1k items the list is always incomplete due to a bug - https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs#L185)

Reply:
oh, I thought it was for VS client only. If we already reporting aggregated completion perf data for vscode, then differentiating between initial and follow-up request might not be that important for now.